nixos/modules/security/acme.nix

```diff
@@ -21,15 +21,41 @@ let
   # The Group can vary depending on what the user has specified in
   # security.acme.certs.<cert>.group on some of the services.
   commonServiceConfig = {
-      Type = "oneshot";
-      User = "acme";
-      Group = mkDefault "acme";
-      UMask = 0022;
-      StateDirectoryMode = 750;
-      ProtectSystem = "full";
-      PrivateTmp = true;
-
-      WorkingDirectory = "/tmp";
+    Type = "oneshot";
+    User = "acme";
+    Group = mkDefault "acme";
+    UMask = 0022;
+    StateDirectoryMode = 750;
+    ProtectSystem = "full";
+    PrivateTmp = true;
+
+    WorkingDirectory = "/tmp";
+
+    CapabilityBoundingSet = [ "" ];
+    DevicePolicy = "closed";
+    LockPersonality = true;
+    MemoryDenyWriteExecute = true;
+    NoNewPrivileges = true;
+    PrivateDevices = true;
+    ProtectClock = true;
+    ProtectHome = true;
+    ProtectHostname = true;
+    ProtectControlGroups = true;
+    ProtectKernelLogs = true;
+    ProtectKernelModules = true;
+    ProtectKernelTunables = true;
+    ProtectProc = "invisible";
+    ProcSubset = "pid";
+    RemoveIPC = true;
+    RestrictAddressFamilies = [
+      "AF_INET"
+      "AF_INET6"
+    ];
+    RestrictNamespaces = true;
+    RestrictRealtime = true;
+    RestrictSUIDSGID = true;
+    SystemCallArchitectures = "native";
+    SystemCallFilter = "@system-service ~@resources ~@ipc ~@keyring ~@resources ~@setuid";
```

Comment:
This is the most controversial part honestly.

As I understand it, in a mixed allow/deny list the evaluation goes from left to right and adds and removes things in order. So I add `@system-service`, then remove `@resources`, `@ipc`, etc. from it.

> If you specify both types of this option (i.e. allow-listing and deny-listing), the first encountered will take precedence and will dictate the default action (termination or approval of a system call). Then the next occurrences of this option will add or delete the listed system calls from the set of the filtered system calls, depending of its type and the default action. (For example, if you have started with an allow list rule for read() and write(), and right after it add a deny list rule for write(), then write() will be removed from the set.)

Honestly I'm now not sure, why `systemd-analyze security` still shows I allow `@privileged`. I'm not worried about `@privileged`, because we actually allow `@chown`.

Reply:
Is it intentional to have `~@resources` there twice? Also how does this affect a call to systemctl (for example to restart another service as configured in `security.acme.certs.<cert>.postRun`)?

Reply:
> Is it intentional to have ~@resources there twice?

Indeed unintentional, thanks for spotting this.

> Also how does this affect a call to systemctl (for example to restart another service as configured in security.acme.certs.<cert>.postRun)?

This is what is in `@resources`, so I think not at all.

```
❯ systemd-analyze syscall-filter @resources
@resources
    # Alter resource settings
    ioprio_set
    mbind
    migrate_pages
    move_pages
    nice
    sched_setaffinity
    sched_setattr
    sched_setparam
    sched_setscheduler
    set_mempolicy
    setpriority
    setrlimit
```

Unless you meant the syscall filter in general. I think `@process`, which is part of `@system-service` takes care of that:

```
❯ systemd-analyze syscall-filter @process
@process
    # Process control, execution, namespacing operations
    arch_prctl
    capget
    clone
    clone3
    execveat
    fork
    getrusage
    kill
    pidfd_open
    pidfd_send_signal
    prctl
    rt_sigqueueinfo
    rt_tgsigqueueinfo
    setns
    swapcontext
    tgkill
    times
    tkill
    unshare
    vfork
    wait4
    waitid
    waitpid
``

Reply:
Remove the duplicate `@resources` deny entry.